### PR TITLE
Add an assertion that incoming UTF8 string is valid

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Assert that consumed can't be zero which might cause infinite loop.

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

**Review by gacholio — APPROVED** [review]

**Comment by gacholio** [comment]
jenkins test sanity zlinux jdk8

File: runtime/gc_base/StringTable.cpp
Comment (dmitripivkine) [inline runtime/gc_base/StringTable.cpp:626]: I believe you can use assertion with message like `Assert_GC_true_with_message`, one of defined in https://github.com/eclipse/omr/blob/master/gc/base/ModronAssertions.h
It provides similar handling as for call in VM https://github.com/eclipse/openj9/blob/868f0ec9bd1a3e10f284f02c1f82703bfb020a29/runtime/vm/jnimisc.cpp#L253 and allows to output more details

**Review by dmitripivkine — APPROVED** [review]
With suggested improvement assertion with message or without it looks good to me

File: runtime/gc_base/StringTable.cpp
Comment (JasonFengJ9) [reply runtime/gc_base/StringTable.cpp]: Sure, improved with `Assert_GC_true_with_message2(MM_EnvironmentBase::getEnvironment(vmThread->omrVMThread), 0 != consumed, "Invalid UTF8 character at %p, %lu", tempData, tempLength)`.

File: runtime/gc_base/StringTable.cpp
Comment (dmitripivkine) [inline runtime/gc_base/StringTable.cpp:626]: We always use `%zu` for `UDATA` type instead of `%lu` is used. I believe it works better for `long long` type

Comment (gacholio) [reply runtime/gc_base/StringTable.cpp:626]: Yes, `z` for UDATA. Are those asserts smart enough to evaluate the assert condition before the other arguments?

Comment (dmitripivkine) [reply runtime/gc_base/StringTable.cpp:626]: Yes it is. Also further handling is done in "prevent inline" style function in .cpp, so it should not increase stack frame size.

Comment (gacholio) [reply runtime/gc_base/StringTable.cpp]: OK let's wait for the `zu` change before merge.

**Comment by dmitripivkine** [comment]
Jenkins compile all jdk11

**Comment by dmitripivkine** [comment]
jenkins test sanity zlinux jdk8

File: runtime/gc_base/StringTable.cpp
Comment (JasonFengJ9) [reply runtime/gc_base/StringTable.cpp]: Fixed.

**Pull request merged**
